src/apic-extension/azext_apic_extension/command_patches.py

```diff
@@ -379,3 +381,65 @@ def pre_operations(self):
 
         args.apim_resource_id = (f"/subscriptions/{subscription_id}/resourceGroups/{resource_group}/providers/"
                                  f"Microsoft.ApiManagement/service/{args.apim_name}/")
+
+        # Set api_source_type
+        args.api_source_type = "AzureApiManagement"
+
+
+@register_command(
+    "apic integration create amazon-api-gateway",
+    is_preview=True,
+)
+class CreateAmazonApiGatewayIntegration(DefaultWorkspaceParameter, CreateIntegration):
+    """Add Amazon API Gateway as API source
+
+    :example: Add Amazon API Gateway as an API source
+        az apic integration create amazon-api-gateway -g contoso-resources -n contoso
+            --integration-id sync-from-my-amazon-api-gateway
+            --access-key https://mykey.vault.azure.net/secrets/AccessKey
+            --secret-access-key https://mykey.vault.azure.net/secrets/SecretAccessKey
+            --region-name us-east-2
+    """
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        # pylint: disable=protected-access
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        # Remove the amazon_api_gateway_source parameter
+        args_schema.amazon_api_gateway_source._registered = False
+        # Remove the apim_resource parameter
+        args_schema.apim_resource_id._registered = False
+        args_schema.msi_resource_id._registered = False
+
+        # Add separate parameters for access-key, secret-access-key, and region-name
+        args_schema.access_key = AAZStrArg(
+            options=["--access-key"],
+            arg_group="AmazonApiGatewaySource",
+            help="Amazon API Gateway Access Key. Must be an Azure Key Vault secret reference.",
+            required=True,
+        )
+        args_schema.secret_access_key = AAZStrArg(
+            options=["--secret-access-key"],
+            arg_group="AmazonApiGatewaySource",
+            help="Amazon API Gateway Secret Access Key. Must be an Azure Key Vault secret reference.",
+            required=True,
+        )
+        args_schema.region_name = AAZStrArg(
+            options=["--region-name"],
+            arg_group="AmazonApiGatewaySource",
+            help="Amazon API Gateway Region (ex. us-east-2).",
+            required=True,
+        )
+        return args_schema
+
+    def pre_operations(self):
+        super().pre_operations()
+        args = self.ctx.args
+        args.api_source_type = "AmazonApiGateway"
+
+        # Set the properties for Amazon API Gateway source
+        args.amazon_api_gateway_source = {
```

Comment:
We may not need to define this object parameter after we flat this object parameters in aaz-dev. You can refer the implementation for APIM sync. This could make our code simpler.

Reply:
I believe this step is like the step of consisting of apim_resource_id. So it is required.